slack-lint/src/main/java/slack/lint/resources/MissingResourceImportAliasDetector.kt

```diff
@@ -0,0 +1,134 @@
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UImportStatement
+import org.jetbrains.uast.USimpleNameReferenceExpression
+import org.jetbrains.uast.getContainingUFile
+import org.jetbrains.uast.getQualifiedParentOrThis
+import slack.lint.resources.model.IMPORT_ALIASES
+import slack.lint.resources.model.RootIssueData
+import slack.lint.util.sourceImplementation
+
+/** Reports an error when an R class, other than the local module's, is imported without an import alias. */
+class MissingResourceImportAliasDetector : Detector(), SourceCodeScanner {
+
+  private val fixes = mutableListOf<LintFix>()
+  private var rootIssueData: RootIssueData? = null
+
+  override fun afterCheckFile(context: Context) {
+    rootIssueData?.let {
+      context.report(
+        ISSUE,
+        it.nameLocation,
+        "Please use an import alias here",
+        quickfixData =
+        fix()
+          .name("Add import alias")
+          // Apply the fixes in reverse so that the ranges/locations don't change.
+          .composite(*fixes.reversed().toTypedArray())
+          .autoFix()
+      )
+      rootIssueData = null
+      fixes.clear()
+    }
+  }
+
+  override fun getApplicableUastTypes(): List<Class<out UElement>> =
+    listOf(UImportStatement::class.java, USimpleNameReferenceExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return object : UElementHandler() {
+
+      override fun visitImportStatement(node: UImportStatement) {
+        // Import alias is a Kotlin feature.
+        val importDirective = node.sourcePsi as? KtImportDirective ?: return
+
+        if (importDirective.importedName?.identifier == "R" && importDirective.aliasName == null) {
+          val packageName = node.getContainingUFile()?.packageName ?: return
+
+          val importedFqName = importDirective.importedFqName
+          val parentImportedFqName = importedFqName?.parent()?.asString()
+
+          val isLocalResourceImport = parentImportedFqName?.let { packageName.startsWith(it) } ?: true
+          if (!isLocalResourceImport) {
+            val importedFqNameString = requireNotNull(importedFqName).asString()
+            val alias = IMPORT_ALIASES[importedFqNameString]
+
+            if (alias != null) {
+              rootIssueData = RootIssueData(alias = alias, nameLocation = context.getNameLocation(importDirective))
+
+              fixes.add(createImportLintFix(importDirective, importedFqNameString, alias))
+            } else {
+              context.report(ISSUE, context.getNameLocation(importDirective), "Please use an import alias here")
```

Comment:
These should be imperative rather than requests
```suggestion
              context.report(ISSUE, context.getNameLocation(importDirective), "Use an import alias for R files from other projects.")
```

Reply:
82f8eaa69eaac58e7a5b4c50f6cd754a9260aad6

Reply:
6434a74c7b8702ddbc04afde02d3ee72c4665559